include/swift/SILOptimizer/Utils/ConstExpr.h

```diff
@@ -40,6 +42,89 @@ class SymbolicValueAllocator;
 class ConstExprFunctionState;
 class UnknownReason;
 
+class StringLiteralInfo {
+  StringLiteralInfo() = default;
+
+public:
+  bool isAscii;
+  StringRef value;
+
+  static Optional<StringLiteralInfo> create(SILValue v) {
+    if (auto *inst = dyn_cast<ApplyInst>(v))
+      return create(inst);
+    return {};
+  }
+
+  static Optional<StringLiteralInfo> create(SILInstruction *inst) {
+    ApplyInst *makeStr = getStringMakeUTF8Apply(inst);
+    if (!makeStr)
+      return {};
+
+    auto strVal = getUTF8String(makeStr);
+    if (!makeStr)
+      return {};
+
+    StringLiteralInfo info;
+    info.value = strVal.getValue();
+    info.isAscii = getIsAscii(makeStr);
+    return info;
+  }
+
+  /// If the given instruction is a call to the compiler-intrinsic initializer
+  /// of String that accepts string literals, return the called function.
+  /// Otherwise, return nullptr.
+  static SILFunction *getStringMakeUTF8Init(SILInstruction *inst) {
```

Comment:
I would be even clearer here about what that function is, what you expect its signature to be, etc.

Reply:
@zoecarver 

My suggestion: constructFromStringMakeUTF8Init(...)?.

Reply:
@gottesmm what do you mean? Are you suggesting that as the function name? I'll add a bit more clarity about the function signature.

Reply:
Maybe `getStringMakeUTF8InitFunction` would be better because this returns a `SILFunction*`, it doesn't construct a `StringLiteralInfo` object.